Makefile

```diff
@@ -180,9 +172,33 @@ else
 EKSCTL_IMAGE_BUILD_ARGS += --build-arg=TEST_TARGET=test
 endif
 
+.PHONY: eksctl-deps-image
+eksctl-deps-image: ## Create a cache image with all the build dependencies
+	@# Pin dependency file permissions.
+	@# Docker uses the file permissions as part of the COPY hash, which can lead to cache misses
+	@# in hosts with different default file permissions (umask).
+	chmod 0600 go.mod go.sum
+	chmod 0700 install-build-deps.sh
```

Comment:
Alternatively, we could tar up the context and pass to docker via stdin, I've seen it done in LinuxKit. Just FYI.

Reply:
Do you have a link?

Reply:
No, LinuxKit has build system written in Go no.

But see the [docs](https://docs.docker.com/engine/reference/commandline/build/), there are multiple options of how you can pass the context.